### DISCUSSION TITLE
Extracting glyphs into plain text

### DISCUSSION THREAD

**Discussion by yaaama** [ryanoasis/nerd-fonts#1160] [Q&A]
Is it possible to extract all of the characters and symbols from a nerd font ttf file? 

So I wanted the symbols and their description to be extracted from the "NerdFontSymbolsOnly" release so that I can keep it in a plain text file and traverse over it with dmenu to search for icons when needed. 

Currently I have been trying to copy them using fontforge but I really don't know what I am doing as font stuff is out of my depth. 

So I was just wondering if that was something that was possible and if I could get some ideas/help on how to do that (if it is possible).

Thanks 🙏

**Comment by Finii** [comment]
Not sure what you want/need.

All icon names are in the [`glyphnames.json`](https://github.com/ryanoasis/nerd-fonts/blob/master/glyphnames.json) file.
You can use `jq` to extract the content in any plain text way you need.

Or do you want the outlines as text (i.e. svgs)?

<img width="523" alt="Screenshot 2023-04-03 at 09 18 51" src="https://user-images.githubusercontent.com/16012374/229438638-278c0e3f-1c3f-49df-bd64-1af67106505e.png">

**Reply by yaaama** [reply]
That is exactly what I was hoping for! I'm not familiar with jq, would you please help me turn the JSON data into something like this:

```
  address-book; f2b9
[glyph] [name of glyph];  [glyph code]
```

Thank you for the help

**Reply by Finii** [reply]
Maybe
```
$ cat glyphnames.json | jq -r 'to_entries[] | "\(.value.char) \(.key); \(.value.code)"'
```

Note that this also create the useless METADATA line, I assume you can delete it ;)

![image](https://user-images.githubusercontent.com/16012374/230047052-35df7008-4b46-44ab-93dc-1ad9c90e3668.png)

**Reply by yaaama** [reply]
You are the best! Thank you so much!
